lib/gc/Transforms/LowerTileVectorPass.cpp

```diff
@@ -0,0 +1,680 @@
+      Operation *producer = fusedOperand->get().getDefiningOp();
+      return producer && producer->hasOneUse();
+    };
+    // Some operation convert as constant, this pattern can help us to improve
+    // the performance.
+    tensor::populateRewriteAsConstantPatterns(patternsInit, defaultControlFn);
+    // Remove unnessary operation like extract slice and insert slice
+    tensor::populateReassociativeReshapeFoldingPatterns(patternsInit);
+    tensor::populateMergeConsecutiveInsertExtractSlicePatterns(patternsInit);
+    tensor::populateFoldTensorSubsetOpPatterns(patternsInit);
+    // Pad operation will lower to linalg.fill. We lower it in init patterns
+    // then lower the fill operation in second patterns.
+    linalg::populatePadOpVectorizationPatterns(patternsInit);
+
+    GreedyRewriteConfig configInit;
+    // Init patterns use to remove useless tensor operation like extract or
+    // insert slice.
+    configInit.strictMode = GreedyRewriteStrictness::ExistingOps;
+    (void)applyPatternsAndFoldGreedily(funcOp, std::move(patternsInit),
+                                       configInit);
+
+    RewritePatternSet firstPatterns(ctx);
+    // All the dynamic shape will reject to lower.
+    populateLowerToTileVectorPatterns(firstPatterns);
+    GreedyRewriteConfig configFirstPn;
+    // We only apply the lowering pattern on existing operations
+    configFirstPn.strictMode = GreedyRewriteStrictness::ExistingOps;
+    (void)applyPatternsAndFoldGreedily(funcOp, std::move(firstPatterns),
+                                       configFirstPn);
+    // Error case:
+    // ```
+    // linalg.copy : <1x32xf32>
+    // tensor.insert_slice tensor<1x32xf32> to tensor<1x128x1x32xf32>
+    // --> lowering as:
+    // transfer_write : permutation map = (d0, d1, d2, d3) -> (d0, d3)
+    // ```
+    // Inorder to avoid the fold greedily bug (fold wrong
+    // permution map for the transfer_write operation). Give it the new full IR
+    // to fold second time can fold correctly.
+    RewritePatternSet secondPattern(ctx);
+    // Ensure each operation has a clear semantics, rather than a composite
+    // semantics. Instead of leaving it to the subsequent passes to handle these
+    // complex semantics, it reduces the difficulty of handling operations in
+    // the subsequent passes. Like transfer_read and transfer_write may have
+    // transpose or braodcast semantic etc.
+    vector::populateVectorTransferPermutationMapLoweringPatterns(secondPattern);
+    // Remove unnessary broadcast operation
+    vector::populateSinkVectorBroadcastPatterns(secondPattern);
+    // Second fold can help us to eliminate redundant operation like consecutive
+    // read and write.
+    (void)applyPatternsAndFoldGreedily(funcOp, std::move(secondPattern));
+    // may need other patterns to reduce redundant operations
+  }
+};
+} // namespace
+
+std::unique_ptr<Pass> createLowerTileVectorPass() {
```

Comment:
I thought this will be auto generated by `.td`?

Reply:
`createLowerToTileVectorPass()`

Reply:
Please check the names for this pass and unify them.

Reply:
Yes, it's redundant. Need to remove it.